__tests__/scripts.test.js

```diff
@@ -38,6 +41,21 @@ test('postinstall stops when detecting unknown shell', async () => {
   expect(containsAlias).toBe(false)
 })
 
+test('postinstall script does not run in yarn', async () => {
+  // yarn does not allow for stdin input during `yarn global add npq`
+  mockIsRunningInYarn.mockRestore()
+  const oldExecPath = process.env.npm_execpath
+
+  const spy = jest.spyOn(console, 'log')
+  process.env.npm_execpath = '/usr/lib/node_modules/yarn/bin/yarn.js'
```

Comment:
interesting test :-)
it probably runs only on some machines but not on all of them. I'm ok with this for now but if you have a quick idea how to make this more generic that's probably best. For example, we can try and detect if that path even exists, if it doesn't to not fail the test.

Reply:
Ah, we're not actually executing that path -- it's just an example path that makes the install script think we're running in yarn.

I just added a commit that clarifies this a bit; let me know if you think I could improve it :)

Reply:
Oh gotcha! makes sense to mock all the way there :-)